scripts/build-rosinstall.py

```diff
@@ -123,16 +123,34 @@ def build_file(fn_bug_desc, overwrite=False):
                        err.returncode, fn_bug_desc)
         return
 
+    # added to the top of the file after processing
+    header = ''
+
+    # ensure catkin >= 0.5.78 (i.e., supports --only-pkg-with-deps)
+    # https://github.com/ros/catkin/commit/913488427d2ff18b808764d1eaf38acead67e18f
+    if not 'catkin' in deps:
+        raise Exception("expected 'catkin' package in .rosinstall file")
+    version_str_catkin = deps['catkin']['version'].split('-')[-2]
+    version_catkin = tuple(map(int, version_str_catkin.split('.')))
```

Comment:
Could consider using [packaging.version.parse(..)](https://packaging.pypa.io/en/latest/version/#packaging.version.parse)?

Reply:
I took a look at `packaging`. It seems quite neat, but I wasn't sure that it was worth introducing another dependency for the sake of a couple of lines. I'll concede that it's much easier to read the code and understand what's going on with `packaging`, so I don't mind using it.